### DISCUSSION TITLE
Right way/endpoint to check if an invoice has been sent via email

### DISCUSSION THREAD

**Discussion by massi82** [fattureincloud/api#406] [Q&A] [answered]
Hello,

as per the subject, I'm looking for a way to understand if am invoice has been sent via email given its document id. I'm not able to find any field such as "status" in the payload of the endpoint for listing the issued document (GET /c/{company_id}/issued_documents). On the other hand I see that there is an endpoint for getting the email data (GET /c/{company_id}/issued_documents/{document_id}/email), but I'm not clear about how to use it to understand if an invoice has been sent or not.

Can anyone please advise?

Thanks in advance.

**Comment by emajo** [comment]
Hi @massi82 to check the status of the email you can check the field `seen_date`: if the field is set to `null` it means that the email has not been sent, if it's set to `1970-01-01` it has been sent but not opened yet by the customer, and if it's set to a different value then it has ben opened at the value's date.

Thanks,
Emanuele.

**Comment by massi82** [comment]
Hi @emajo,

thanks for the reply. I've found the field seen_date in the "List Issued Documents" payload, but in our case it is always to null, even for the invoices that have been successfully sent (as properly shown in the GUI). Is there anything I'm missing? 

Thanks for the help.

**Reply by valmoz** [reply]
Hi @massi82 are you using the fieldset "detailed"?

Thanks
Mauro

**Reply by massi82** [reply]
Yes, in fact I can successfully find the field seen_date, but it is always equal to null, as I can easily verify in the API playground. In fact this is an example of an invoice with due date 2023-11-30 (that has clearly already been sent), bu seen_date is null. Any help is appreciated.

![image](https://github.com/fattureincloud/api/assets/14828435/daa211b7-0cb3-4a28-881c-ca96133415a5)

**Reply by emajo** [reply]
Are you sure it has been sent via email? Did you do it using the [Schedule Email](https://developers.fattureincloud.it/api-reference#post-/c/-company_id-/issued_documents/-document_id-/email) api endpoint or the web app?

**Reply by massi82** [reply]
Yes, here is the screenshot of the status of the invoice:

![image](https://github.com/fattureincloud/api/assets/14828435/6e8dff63-d39f-4a06-92b2-5a5df21f7ed8)

It has been created via the web app. By a quick check it looks like all our invoices have that field set to null, regardless of having been sent or not.

**Reply by emajo** [reply, marked as answer]
That is not an invoice sent by email, it has been sent to the SDI, in this case the field you need to check is `ei_status`, you can find all the possible values on our documentation.

**Reply by massi82** [reply]
Ahhh, got it, thank you for the answer. So in my case I need to use the POST /c/{company_id}/issued_documents/{document_id}/e_invoice/send endpoint instead of the email one.

Thanks for the help!
